### ISSUE TITLE
Unable to log in to addons.mozilla.org

### ISSUE THREAD

**Issue by lovasoa** [mozilla/addons-server#18672]
Hello !

I was not sure where to report this issue, so please feel free to close it if it is not the right place.

### Describe the problem and steps to reproduce it:

1. I open https://addons.mozilla.org
2. I click "log in"
3. If I open my dev tools, I see the following URLs being loaded:
    
![image](https://user-images.githubusercontent.com/552629/150651950-c3ee2597-5e93-45d7-8a52-51caca1f783c.png)

4. I am redirected to the original page (https://addons.mozilla.org/en-US/firefox/) without being logged in.

### Expected result

After the final redirection, I should be logged in addons.mozilla.org 

### Actual result

I am not logged in. The page header still displays a "log in" button

![image](https://user-images.githubusercontent.com/552629/150652036-f4964425-c159-4f50-be80-b1b23d616b84.png)

### Anything else we should know?

 I reported it independently to firefox accounts: https://github.com/mozilla/fxa/issues/11708

**Comment by bluedaze** [comment]
What OS are you on? I am unable to recreate this issue on Linux. Maybe this is windows specific.

**Comment by lovasoa** [comment]
I am on linux too (fedora 35, Firefox 96).

I just tried again, and now I am blocked https://accounts.firefox.com/inline_totp_setup , which refuses the google authenticator's one-time passwords.

The issue on AMO seems to be that it doesn't handle errors in the oauth callback correctly. If you look at my screenshot above, I think the problem is here:

![image](https://user-images.githubusercontent.com/552629/150704802-4346c23c-1107-40cb-a3f1-a1b8dd310d6a.png)

This page on amo should probably have displayed an error message instead of redirecting to the home page.

**Comment by diox** [comment]
Yes, AMO doesn't show you the error, but ultimately that wouldn't help you much: there is a problem on FxA's side with your account/the way you're authenticating for some reason.

**Comment by lovasoa** [comment]
I think just silently ignoring errors is not a good thing. How is a normal user supposed to react when they click "log in", they successfully enter their credentials on FxA, they get redirected to amo, and they are not logged in ? 

If something went wrong during the authentication process, you cannot really continue as if nothing happened...

**Comment by diox** [comment]
I agree AMO not showing the error is bad UX. I'm just saying that wouldn't help you much here, you'd still be stuck. 

FWIW we log when that happens, and this is why fixing this issue on the AMO side has never been a high priority: there is an extremely low number of people affected by this.

**Comment by lovasoa** [comment]
:+1: In the meantime, is there a way for me to access my account on AMO ?

**Comment by diox** [comment]
We can't do anything from AMO's side since FxA is preventing you from logging in. You'll need to get someone from FxA to look at https://github.com/mozilla/fxa/issues/11708

**Comment by lovasoa** [comment]
Okay. Do you know someone I can ping about this issue ? 

I cannot monitor or update my extensions because of that.

**Comment by diox** [comment]
Since you already filed an issue, hopefully someone will pick it up. Otherwise you can try https://chat.mozilla.org/#/room/#fxa:mozilla.org

**Comment by Madriix** [comment]
I may have managed to correct the problem but I don't know if it's thanks to the thing I modified, or if it's just a coincidence.

Here is what I did to be validated by the form (but no idea if it's a coincidence):
1) Go to https://addons.mozilla.org/en/firefox/ and click on Login
2) Maybe you will have to type your Email and password (not always it seems)
3) Then it asks for the two-factor verification, open your mobile and take the QR CODE in photo
4) Type the code received, you have seen there is an error it does not want to validate the code, it does not want!
5) I had the idea to open Dev tools to edit the html by selecting the input (<input ...) and I removed the length limitation which was something like 6 , there was also this "regex="..."' that I removed, and I emptied "area" and then you validate the code received by Google Authentication and immediately it offers to download or copy the 6 or 9 codes received , you must then retype one of cs 6 or 9 codes in a new input and the account connects.

Here, no idea if it's a coincidence or if it's because I modified the input via the Dev tools

**Comment by Madriix** [comment]
@wagnerand The computer is on time

**Comment by KevinMind** [comment]
Old Jira Ticket: https://mozilla-hub.atlassian.net/browse/ADDSRV-73
